contracts/auth_proxy.rs

```diff
@@ -238,14 +316,8 @@ impl AuthProxyContract {
         let mut tx_json_string = serde_json::to_string(&tx)
             .unwrap_or_else(|e| panic!("Failed to serialize NearTransaction: {:?}", e));
 
-        // Convert any large deposit numbers to strings in the JSON
-        let modified_tx_string = deposits.iter().fold(tx_json_string, |acc, deposit| {
-            acc.replace(
-                &format!("\"deposit\":{}", deposit.0),
-                &format!("\"deposit\":\"{}\"", deposit.0),
-            )
-        });
-        tx_json_string = modified_tx_string;
+        // Convert large deposit numbers to strings for JSON compatibility
+        tx_json_string = self.convert_deposits_to_strings(tx_json_string, &deposits);
```

Comment:
Not sure what's the pattern or expense in smart contracts but I'd anticipate we deserialize into a well structured type at first point of input into the contract's call and serialize when outputting. 
This will also scale better to other future needs to edit the payload.

Why not convert the deposit value into strings in the tx struct before serializing it to json in ln316?

Reply:
@RyanSept If I understand your comment correctly, we are already doing this when the string `actions_json` is [serialized into a list](https://github.com/beneviolabs/ft-allowance-agent/pull/143/files#diff-356e5d5e2dc6d7329d8d8c7b14ff38d5437618f56a5e58ec2ad19d28f54269c9R257) of `ActionString`. Because of our nested input params, json that contains a string value that represents deeply nested JSON, I think we're stuck with this approach of converting a string into JSON, then recurring into that json to ensure its data types are in in the expected format, in this case, converting deposit values into strings without losing accuracy.

Reply:
Ah, I think I see that you're asking why not convert [here](https://github.com/beneviolabs/ft-allowance-agent/pull/143/files#diff-356e5d5e2dc6d7329d8d8c7b14ff38d5437618f56a5e58ec2ad19d28f54269c9R164).  That is a keen observation

Reply:
Perhaps we can avoid creating this `tx_json_string` in favor of passing the serialized `NearTransaction` directly to our callback. I'll time box giving this a try.